msteams-platform/bots/how-to/conversations/bot-copilot-handoff.md

```diff
@@ -0,0 +1,178 @@
+
+You can enable Copilot handoff to allow users to continue the conversation with a custom Copilot from the Copilot for Microsoft chat without losing the context of their work. You can improve your plugins with the Copilot handoff feature, which allows for a seamless transition to your custom copilot by using deep links that carry over the user context. This resolves the issue of users having to restart and repeat their requests in the bot experience. For example, in the following image, the user is using Copilot to get information on tech issues and receives results from across the enterprise dataset. The user doesn't find the information sufficient and wants to continue an interaction with an Aisera bot.​ When the user selects the Aisera bot, a new chat is opened to continue the conversation with Aisera. The continuation of the conversation without losing the context is called Copilot handoff.
+
+:::image type="content" source="../../../assets/images/Copilot/Copilot-handoff.gif" alt-text="The GIF shows the conversation handoff between the Copilot for Microsoft 365 and the Aisera chat bot.":::
+
+## How Copilot handoff works
+
+A custom Copilot sends a deep link with a continuation token to the Copilot for Microsoft 365. The deep link query parameter with a continuation token ensures that any information from your plugin invocation parameters is referenced.  When the user selects the deep link, the Copilot then sends an invoke call to the bot with the continuation token, and the bot can resume the conversation based on the context. This process enables a seamless transition from Copilot to a custom Copilot, maintaining the conversation’s continuity and context, which optimizes the user experience.
+
+:::image type="content" source="../../../assets/images/Copilot/copilot-handoff-flow.png" alt-text="Screenshot shows the handoff flow between the user, Copilot, plugin, Teams, and bot." lightbox="../../../assets/images/Copilot/copilot-handoff-flow.png":::
+
+You must create a deep link URL with a `continuation` query parameter for the action button and assign a continuation token to the parameter to facilitate the handoff process. When the user selects the action button, Microsoft Teams reads the continuation token from the URL and initiates an invoke call to the bot. The bot or plugin uses the continuation token to create a response, which is then displayed to the user in the plugin chat window.
+
+## Enable copilot handoff
+
+To enable Copilot handoff in Teams, follow these steps:
+
+1. **Configure a deep link URL**: Create a deep link to the bot chat and include `botMri` and `continuationToken`. The deep link must be in the format `https://teams.microsoft.com/l/chat/0/0?users=${botMri}&continuation=${continuationToken}`.
+
+   **Example**:
+
+   ```JSON
+   { 
+
+   "type": "Action.OpenUrl", 
+
+   "title": "Handoff to Bot", 
+
+   "url": "https://teams.microsoft.com/l/chat/0/0?users=${botMri}&continuation=${continuationToken}" 
```

Comment:
> https://teams.microsoft.com/l/chat/0/0?users=${botMri}

This deeplink format isn't documented, please document (the docs do not say that you can use botMri for users).

If I remember correctly it was previously documented, but then we removed it because we didn't want using this deep link to start a chat with the bot. Have we changed our mind, and this is ok again?

Reply:
Updated as Create a [deep link to a chat](https://learn.microsoft.com/en-us/microsoftteams/platform/concepts/build-and-test/deep-link-teams#configure-deep-link-to-start-a-chat-manually) and add the `28:<botId>` and `continuationToken` to the deep link format."